src/app/_components/chats/ChatInput.tsx

```diff
@@ -56,21 +70,43 @@ export default function ChatInput({ chatId }: Props) {
         },
       );
     }
+    form.reset({ message: "", imageUrl: "" });
+  };
+
+  const onEnter = (e: React.KeyboardEvent) => {
+    if (e.code == "Enter" && e.shiftKey == false) {
+      e.preventDefault();
+      formRef.current?.requestSubmit();
+    }
+  };
 
-    form.reset({ message: "" });
+  const onUpload = (url: string) => {
+    form.setValue("imageUrl", url);
+    formRef.current?.requestSubmit();
   };
 
+  useEffect(() => {
+    form.register("imageUrl", { required: false });
+  }, [form.register]);
+
+  useEffect(() => {
+    console.log(form.formState.errors);
+  }, [form.formState.errors]);
+
```

Comment:
I believe both of them are redundant, if not explain me and we'll think together of a solution without using use effect

Reply:
I removed them